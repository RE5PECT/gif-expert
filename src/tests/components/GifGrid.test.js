import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test on GifGrid Component', () => {
  const category = 'RickandMorty';

  test('should show the component GifGrid', () => {
    useFetchGifs.mockReturnValue({
        data: [],
        loading: false,
      });
    let wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should de mostrar items cuando se cargan imagenes useFetchGifs', () => {
   
    const gifs = [
      {
        id: 'asdAsdadfa21',
        url: 'https://localhost/asdasd/img.jpg',
        title: 'Cualquier cosa',
      },
    ];
   
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    let wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
