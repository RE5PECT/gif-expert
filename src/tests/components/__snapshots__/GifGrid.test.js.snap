// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGrid Component should show the component GifGrid 1`] = `
<Fragment>
  <h3>
    RickandMorty
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
